src/Hydrators/ProjectHydrator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace ProjectManagementApi\Hydrators;
+
+use DateTime;
+
+use ProjectManagementApi\DatabaseConnection; 
+use ProjectManagementApi\Entities\Project; 
+
+class ProjectHydrator
+{
+    public static function getAllProjects(DatabaseConnection $db)
+    {
+        $query = $db->query('SELECT id, `name`, client_id, deadline FROM projects');
+        $projectsArray = $query->fetchAll();
+        $projectObjectsArray = [];
+        foreach ($projectsArray as $projectItem) {
+            $projectObject = new Project(
+                $projectItem['id'],
+                $projectItem['name'],
+                $projectItem['client_id'],
+                is_null($projectItem['deadline']) ? null : new DateTime($projectItem['deadline'])
```

Comment:
- the database has dates in SQL date format which is basically a string of 'yyyy-mm-dd'
- the API documentation has dates in format 'dd/mm/yyyy'
- you have resolved this by using a DateTime object, which is good
- the constructor of Project therefore takes a DateTime object
- this point is a little bit arguable, but i would say that it would be better if:
  - the constructor takes a string in the format that the DB provides
  - the logic of creating a DateTime object belongs inside the Project class, because it's the Project's own concern
  - that way the Project class is more encapsulated, rather than depending on other classes to satisfy its fussy requirements

Reply:
done!